### ISSUE TITLE
iOs Actions not syncing

### ISSUE THREAD

**Issue by wtangofoxtrot** [home-assistant/iOS#2901]
**iOS device model, version and app version**

Model Name: 13mini and 14
Software Version: 17.5.1
App version: 2024.730

**Home Assistant Core Version**

2024.7.2

**Describe the bug**
iOs Actions are setup on the HA server in the configuration.yaml and have been working for atleast 1 year until they stopped ~2weeks ago. No changes were made on the server but all connected iOs devices show an empty list for synced actions as shown in the below screenshot:

**To Reproduce**
- go to Companion app settings >Actions and tap Update server Actions
- Nothing gets synced across from the server

**Expected behavior**
- ios actions are setup in the server's configuration.yaml using https://companion.home-assistant.io/docs/core/actions/ 
- these actions were previously working but stopped across all connected devices by themselves with no server side changes
- re-added dummy action in the configuration.yaml using above documentation and it too did not sync across after restarting the HA server to load the configuration.yaml

**Screenshots**

![353930568-90dfd2af-21a9-475c-8b13-b0651823c2a5](https://github.com/user-attachments/assets/bb6a4177-2a91-4b35-8985-d3781010bc2c)
**Additional context**

Had filed https://github.com/home-assistant/core/issues/122968 but was told to post here as my testing by reverting the HA server to when it was working did not fix the issue indicating this is likely a mobile app issue.
### Example YAML snippet

```yaml
ios:
  actions:
    - name: Fred
      label:
        text: "Hello, World"
      icon:
        icon: earth
      show_in_carplay: false
      show_in_watch: true
    - name: Desk Toggle
      label:
        text: "WTangoFoxtrot's Desk"
      icon:
        icon: Desk
      show_in_watch: true
```
ios companion app logs: [2024-08-02_15.46.20-0700.logs.zip](https://github.com/user-attachments/files/16475912/2024-08-02_15.46.20-0700.logs.zip)

**Comment by Korvorkian** [comment]
First time using GitHub here but can confirm this is also happening on my Gen1 iPad Pro on 2024.7

**Comment by bgoncal** [comment]
Please share your logs and YAML, I'll check on Monday

**Comment by wtangofoxtrot** [comment]
done @bgoncal

**Comment by bgoncal** [comment]
In your logs it says the iOS component is not loaded in your HA server, if you go to Settings >> Devices & services, do you see "iOS" as one of the integrations there?

Before we move forward let's try this:
- Remove the actions from your configuration.yaml
- Restart HA
- Open App and go to the Actions screen inside companion App settings

- Add actions back to configuration.yaml
- Restart 
- Open App and go to the Actions screen inside companion App settings

**Comment by wtangofoxtrot** [comment]
@bgoncal you're right, the iOs integration had been disabled by me for a different reason much before the issue happened and I think when i updated HA core it reset the actions.

Perhaps the docs for the companion app should be updated to reflect this as that would have definitely helped here, I can push up something to https://companion.home-assistant.io/docs/core/actions/ 

Also would it make sense to have the companion app show the above error or a message that the integration is required for actions ?

**Comment by bgoncal** [comment]
Happy to hear that we found the root causes, is it working for you now? 

Indeed that's something iOS App could alert the user, I'll add to the backlog.

**Comment by pw1124** [comment]
Just want to say that I've had the exact same problem for the past week or so and this thread has been super helpful in getting my issue fixed.

I had also disabled the iOS integration - what threw me off was that I had upgraded my phone recently, and when I was doing some housekeeping tidying up my integrations, I saw only a couple of entities relating to my old phone inside the Home Assistant iOS integration.  Seeing that my new phone was showing up inside the Mobile App integration, I assumed the iOS integration wasn't needed anymore and disabled it, and I had no issues until a week later when I updated my actions in configuration.yaml and tried to sync on the companion app and all the actions were gone.

I think right now maybe it's not super obvious in what the iOS integration and the Mobile App integration are each responsible for, and extra error messages would have definitely helped here!
